lib/src/injector_base.dart

```diff
@@ -97,11 +97,7 @@ class Injector {
     final factory = _factoryMap[identity];
     final factoryId = factory.hashCode;
 
-    if (_factoryCallIds.contains(factoryId)) {
-      throw CircularDependencyException(type: T.toString());
-    }
-
-    _factoryCallIds.add(factoryId);
+    _factoryCallIds.add(factoryId) || (throw CircularDependencyException(type: T.toString()));
```

Comment:
sneaky that you used the `||`  as a control flow statement 🙈 
I would still prefer an if here for readability 👍

Reply:
I did that to avoid comparing with false. Would you prefer `!` or ` == false`? IMO both are rather ugly.

Reply:
I prefer `!` 🙈